src/Components/ADT3DSceneBuilder/Internal/Behaviors/Widgets/WidgetBuilders/DataHistoryWidgetBuilder/DataHistoryWidgetBuilder.tsx

```diff
@@ -0,0 +1,360 @@
+                                        href={DOCUMENTATION_LINKS.dataHistory}
+                                    >
+                                        {t('learnMore')}
+                                    </Link>
+                                </>
+                            )
+                        }}
+                    />
+                </Stack>
+            );
+        },
+        [t]
+    );
+
+    const onChartOptionChange = useCallback(
+        (
+            optionKey: ChartOptionKeys,
+            value:
+                | IDataHistoryChartYAxisType
+                | IDataHistoryAggregationType
+                | number
+        ) => {
+            updateWidgetData(
+                produce(formData, (draft) => {
+                    draft.widgetConfiguration.chartOptions[
+                        optionKey as string
+                    ] = value;
+                })
+            );
+        },
+        [updateWidgetData, formData]
+    );
+
+    const theme = useTheme();
+    const customStyles = getWidgetFormStyles(theme);
+    const actionButtonStyles = getActionButtonStyles(theme);
+    return (
+        <div className={customStyles.widgetFormContents}>
+            <Stack tokens={{ childrenGap: 8 }}>
+                <TextField
+                    required
+                    placeholder={t(
+                        'widgets.dataHistory.form.connectionStringPlaceholder'
+                    )}
+                    label={t('widgets.dataHistory.form.connectionStringTitle')}
+                    value={connectionString}
+                    readOnly
+                    disabled
+                    title={connectionString}
+                    onRenderLabel={handleOnRenderConnectionStringLabel}
+                />
+                <TextField
+                    required
+                    placeholder={t(
+                        'widgets.dataHistory.form.displayNamePlaceholder'
+                    )}
+                    label={t('displayName')}
+                    value={formData.widgetConfiguration.displayName}
+                    onChange={onDisplayNameChange}
+                />
+                <TimeSeriesList
+                    series={formData.widgetConfiguration.timeSeries}
+                    onSeriesEditClick={handleTimeSeriesEditClick}
+                    onSeriesRemoveClick={handleTimeSeriesRemoveClick}
+                />
+                {formData.widgetConfiguration.timeSeries.length <
+                    MAX_NUMBER_OF_TIME_SERIES && (
+                    <ActionButton
+                        id={addTimeSeriesCalloutId}
+                        styles={actionButtonStyles}
+                        text={t('widgets.dataHistory.form.timeSeries.add')}
+                        onClick={toggleIsAddTimeSeriesCalloutVisible}
+                    />
+                )}
+                <Label className={customStyles.label} id={yAxisLabelId}>
```

Comment:
I need to render a tooltip for this label but since there is no "onRenderLabel" prop for `ChoiceGroup `component in fluent this is the suggested customization method.

Reply:
how odd. That must be one of the few that doesn't let you do that. :(